### ISSUE TITLE
Change bulk imports to merge tags

### ISSUE THREAD

**Issue by sanmadjack** [shish/shimmie2#651]
The bulk imports support using folders to tag images, but if the file uses the same file name format that shimmie saves files as it uses only the tags it can extract from the file name, ignoring the folders. 
This behavior seems un-intuitive to me (since it isn't stated in any of the bulk import pages), and less useful than merging all of the potential tags. I have a working version of the code to merge the tags that I can create a merge request for if you approve this change.

**Comment by shish** [comment]
Merging tags from folders + filename sounds sensible :)

**Comment by sanmadjack** [comment]
I switched over to the develop branch, since that seems to be where the work goes these days. In the process I ran into several issues, I've created pull request #653 to address them so that I could get this pull request ready to go.

While setting it up I had an additional thought about how cron upload handles things. If a user sets up folders for tagging items and there is some sort of failure to import, the files are moved to the failed_to_upload folder. This effectively discards all of the tag information, leaving it up to the user to figure out which file was in what folder when (and if) they are able to resolve whatever the issue with the upload was. I think it would be better for imported/failed files to be moved to the uploaded/failed_to_upload folders mirroring the same folder path the file came from. 

So if a file came from queue/1/1/image.bmp and failed, it would end up in failed_to_upload/1/1/image.bmp. That way the tagging info is preserved, and once the upload issue is resolved the user can just move the folder structure back to queue for re-processing. Let me know if this would be desirable for the pull request.
